geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -9753,6 +9753,7 @@ VersionedObjectList basicRemoveAll(final Collection<Object> keys,
           txException = txException.getCause();
         }
         if (runtimeException == null) {
+          runtimeException = getCancelCriterion().generateCancelledException(e.getFailure());
```

Comment:
why do you have two assignments of "runtimeException"? Seems like the second assignment just throws away your new one.

Reply:
You are right, it should not set runtimeException as getCancelCriterion().generateCancelledException would throw exception if cache is closing. I was just use the code in basicPutAll.

Reply:
Actually, the method may not throw CacheClosedException but return CacheClosedException instead. Made the change necessary.